src/components/Attachments/AttachmentView/AttachmentViewPdf/index.android.js

```diff
@@ -15,17 +15,27 @@ function AttachmentViewPdf(props) {
     const offsetX = useSharedValue(0);
     const offsetY = useSharedValue(0);
 
+    // Reanimated freezes all objects captured in the closure of a worklet.
+    // Since Reanimated 3, entire objects are captured instead of just the relevant properties.
+    // See https://github.com/software-mansion/react-native-reanimated/pull/4060
+    // Because context contains more properties, all of them (most notably the pager ref) were
+    // frozen, which combined with Reanimated using strict mode since 3.6.0 was resulting in errors.
+    // Without strict mode, it would just silently fail.
+    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze#description
+    const isContextReady = attachmentCarouselPagerContext !== null;
+    const shouldPagerScroll = isContextReady ? attachmentCarouselPagerContext.shouldPagerScroll : undefined;
+
     const Pan = Gesture.Pan()
         .manualActivation(true)
         .onTouchesMove((evt) => {
-            if (offsetX.value !== 0 && offsetY.value !== 0 && attachmentCarouselPagerContext) {
+            if (offsetX.value !== 0 && offsetY.value !== 0 && isContextReady && shouldPagerScroll !== undefined) {
```

Comment:
why not just undefined? 
```suggestion
            if (offsetX.value !== 0 && offsetY.value !== 0 && shouldPagerScroll) {
```

Reply:
Ok changed, looks good now?

Reply:
yes, looks good! As this is dev-only crash, I am not sure if this change should be added in https://github.com/Expensify/App/pull/34078 as urgent fix or leave as is for now and handle it in this RN upgrade PR.
@thienlnam or @NikkiWines will decide. (coming from https://expensify.slack.com/archives/C01GTK53T8Q/p1704712565926769?thread_ts=1704683824.693909&cid=C01GTK53T8Q)

Reply:
> production code (dev build)

Ah ok, thanks for clarifying. In that case, it's less urgent and can go out in this PR - so long as we update the various image/pdf related issues to note that a fix for the dev-only crashes is handled here.